srml/support/src/dispatch.rs

```diff
@@ -906,10 +913,13 @@ macro_rules! __dispatch_impl_metadata {
 		$($rest:tt)*
 	) => {
 		impl<$trait_instance: $trait_name> $mod_type<$trait_instance> {
+			pub fn call_module() -> $crate::dispatch::CallMetadata {
+				__call_to_metadata!($($rest)*)
+			}
 			pub fn metadata() -> $crate::dispatch::ModuleMetadata {
 				$crate::dispatch::ModuleMetadata {
 					name: $crate::dispatch::DecodeDifferent::Encode(stringify!($mod_type)),
-					call: __call_to_metadata!($($rest)*),
+					call: Self::call_module(),
```

Comment:
Why this change?

Reply:
it is probably a remnant from the other version (when I kept the code from previous encoding), does not make sense anymore

Reply:
mistake, it is used at another place